src/common/NetworkPrivate.cpp

```diff
@@ -3,16 +3,115 @@
 #include "common/NetworkManager.hpp"
 #include "common/NetworkResult.hpp"
 #include "common/QLogging.hpp"
-#include "debug/AssertInGuiThread.hpp"
 #include "singletons/Paths.hpp"
 #include "util/DebugCount.hpp"
 #include "util/PostToThread.hpp"
 
+#include <magic_enum/magic_enum.hpp>
 #include <QCryptographicHash>
 #include <QFile>
 #include <QNetworkReply>
 #include <QtConcurrent>
 
+#ifndef NDEBUG
+#    define CHATTERINO_WARN_SLOW_HTTP
+#endif
+
+#ifdef CHATTERINO_WARN_SLOW_HTTP
+#    include <QElapsedTimer>
+
+#    ifdef NDEBUG
+constexpr qsizetype SLOW_HTTP_THRESHOLD = 30;
+#    else
+constexpr qsizetype SLOW_HTTP_THRESHOLD = 90;
+#    endif
+#endif
+
+namespace {
+
+using namespace chatterino;
+
+/// Guard to call `deleteLater` on a QObject when destroyed.
```

Comment:
Maybe this should be moved somewhere else? It feels weird lying next to network internals.